miner/scroll_worker.go

```diff
@@ -650,14 +652,30 @@ func (w *worker) handlePipelineResult(res *pipeline.Result) error {
 	}
 
 	var commitError error
-	if res != nil && res.FinalBlock != nil {
+	if res.FinalBlock != nil {
 		if commitError = w.commit(res); commitError == nil {
 			return nil
 		}
 		log.Error("Commit failed", "header", res.FinalBlock.Header, "reason", commitError)
+		if _, isRetryable := commitError.(retryableCommitError); !isRetryable {
+			return commitError
+		}
 	}
 	w.startNewPipeline(time.Now().Unix())
-	return commitError
+	return nil
+}
+
+// retryableCommitError wraps an error that happened during commit phase and indicates that worker can retry to build a new block
```

Comment:
Maybe define a error

retryableCommitError := errors.New("xxxxxxxxx)

```
if err = w.engine.VerifyHeader(w.chain, block.Header(), true); err != nil {
		return retryableCommitError
	}
```

```
if errors.Is(commitError, retryableCommitError) {
   return commitError
}
```

Reply:
I think I type is more suitable since it allows access to the underlying error as well.

Reply:
```
// You can edit this code!
// Click here and start typing.
package main

import (
	"errors"
	"fmt"
)

func main() {
	// origin error
	originErr := errors.New("this is origin err")
	wrapErr := fmt.Errorf("wrap err: %w", originErr)

	fmt.Println("wrap err--->: ", wrapErr)
	fmt.Println("get origin err from wrap err: ", errors.Unwrap(wrapErr))
}
```
If want get underlying error maybe `errors` more convenient